### DISCUSSION TITLE
Variable naming convention

### DISCUSSION THREAD

**Discussion by NiedziolkaMichal** [mdn/mdn-community#307] [Content]
[Java Script code style guide](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#variable_names) provides hints about variable naming, that I believe are too broad. There are plenty of pages in which different names are used for the same structures. On page [Array.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) different examples use names `array1`, `array` and `myArray` for an ordinary array. In all JS content examples, the name `arr` is used 75 times, `myArray` 23 times, `array` 22 times, while most of the interactive examples use the name `array1`. Map object is usually called `myMap`, `map`, or `map1`.

I think the style guide should provide a list of fallback names, for each primitive value and data structure. Those names should be used, only when more expressive names like `cars` or `numbers` are not suitable. This convention should be also used for interactive examples, which have [a separate style guide](https://github.com/mdn/interactive-examples/blob/main/JS-Example-Style-Guide.md).

**Comment by Josh-Cena** [comment]
I personally use `array`/`generator` (the constructor function, first letter lowercased) within prose to denote instance properties. For example:

> This method accesses `array.length`.
> You can call `generator.next()`.

But such cases are rare, as you can usually omit the prefix and say "accesses the `length` property".

For code examples, I prefer the name that looks less like it was chosen on purpose. `array` add an identifier sounds like there's something special about it. Therefore I usually go with `arr`, `gen`, etc. `myArray` is unnecessary, but `myMap` is necessary as an arbitrary identifier, because `map` looks too special while there's no abbreviation of `Map`.

However, in general my experience is that strict naming conventions are really hard to enforce. We can at best tell people to prefer X over Y, but it's very likely that Y still keeps creeping into the content over time.

**Comment by scffs** [comment]
I agree with you
It sounds useful
